docs/source/name_inference.rst

```diff
@@ -0,0 +1,428 @@
+.. currentmodule:: torch
+
+.. _name_inference_reference-doc:
+
+Named Tensors operator coverage
+===============================
+
+Please read :ref:`named_tensors-doc` first for an introduction to named tensors.
+
+This document is a reference for *name inference*, a process for how named tensors
+use names to automatically check that APIs are being called correctly at runtime and
+for how operators propagate names from input tensors to outputs.
+
+More formally, name inference consists of the following two steps:
```

Comment:
I don't think this restatement is helpful.  You could just put the dimension matching as a specific correctness check above.